From 90583b3a304c88e53e79e51e0658b15318b18244 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:08 -0400
Subject: [fs] nfs: Handle allocation errors correctly in objlayout_alloc_layout_hdr()

Message-id: <1403798077-28799-50-git-send-email-steved@redhat.com>
Patchwork-id: 84950
O-Subject: [RHEL7.1 PATCH 49/78] pNFS: Handle allocation errors correctly in objlayout_alloc_layout_hdr()
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Return the NULL pointer when the allocation fails.

Cc: Boaz Harrosh <bharrosh@panasas.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 0aa61e78a0f262a2f94bd138831c97749cfca5bf
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/objlayout/objlayout.c b/fs/nfs/objlayout/objlayout.c
index 2f955f6..765d3f5 100644
--- a/fs/nfs/objlayout/objlayout.c
+++ b/fs/nfs/objlayout/objlayout.c
@@ -53,10 +53,10 @@ objlayout_alloc_layout_hdr(struct inode *inode, gfp_t gfp_flags)
  struct objlayout *objlay;
 
  objlay = kzalloc(sizeof(struct objlayout), gfp_flags);
- if (objlay) {
-  spin_lock_init(&objlay->lock);
-  INIT_LIST_HEAD(&objlay->err_list);
- }
+ if (!objlay)
+  return NULL;
+ spin_lock_init(&objlay->lock);
+ INIT_LIST_HEAD(&objlay->err_list);
  dprintk("%s: Return %p\n", __func__, objlay);
  return &objlay->pnfs_layout;
 }
-- 
1.7.1